internal/envoy/v3/config_gen.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+
+	"github.com/projectcontour/contour/internal/timeout"
+)
+
+// DefaultXDSClusterName is the default name of the Contour cluster as defined in the
+// Envoy bootstrap configuration.
+const DefaultXDSClusterName = "contour"
+
+type EnvoyGen struct {
+	xdsClusterName string
+}
+
+type EnvoyGenOpt struct {
+	XDSClusterName string
+}
+
+func NewEnvoyGen(opt EnvoyGenOpt) *EnvoyGen {
+	return &EnvoyGen{
+		xdsClusterName: opt.XDSClusterName,
```

Comment:
WDYT about having this constructor set default values for fields like this if not provided by the caller? That way we could avoid having to pass `DefaultXDSClusterName` every place this constructor is used. Of course if the caller provides a non-empty value, that would be used instead of the default. Something like:

```go
func valOrDefault(val, defaultVal string) string {
	if val != "" {
		return val
	}
	return defaultVal
}

func NewEnvoyGen(opt EnvoyGenOpt) *EnvoyGen {
	return &EnvoyGen{
		xdsClusterName: valOrDefault(opt.XDSClusterName, DefaultXDSClusterName),
	}
}
```

Reply:
The reason I didnt do that is because the empty value to take a different meaning with https://github.com/projectcontour/contour/pull/6806

Reply:
Happy to change it, up to you. But I thought that if `xdsClusterName` is `nil` we would want to switch to using a different `configSource` config.

I guess you can argue that in that case having the default might be better but that was my motivation

Reply:
OK, that's fair. We can leave it as is, easy enough to make any changes down the road anyway.